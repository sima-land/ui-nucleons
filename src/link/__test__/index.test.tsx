import { mount } from 'enzyme';
import { Link } from '..';

describe('<Link />', () => {
  it('calls helpers with right params and renders correctly without external', () => {
    const wrapper = mount(
      <Link className='testClass' href='/cart/' color='basic-gray38' target='_blank'>
        Test link
      </Link>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('calls helpers and renders correctly with pseudo', () => {
    const wrapper = mount(
      <Link pseudo color='basic-gray38'>
        Test link
      </Link>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('calls helpers and renders correctly with external', () => {
    const wrapper = mount(
      <Link href='/cabinet/' color='basic-blue'>
        Test link
      </Link>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "onClick/onMouseEnter/onMouseLeave..." props', () => {
    const onClickSpy = jest.fn();
    const onMouseEnterSpy = jest.fn();
    const onMouseLeaveSpy = jest.fn();

    const wrapper = mount(
      <Link onClick={onClickSpy} onMouseEnter={onMouseEnterSpy} onMouseLeave={onMouseLeaveSpy}>
        Test link
      </Link>,
    );

    wrapper.simulate('click');
    expect(onClickSpy).toHaveBeenCalledTimes(1);

    wrapper.simulate('mouseEnter');
    expect(onMouseEnterSpy).toHaveBeenCalledTimes(1);

    wrapper.simulate('mouseLeave');
    expect(onMouseLeaveSpy).toHaveBeenCalledTimes(1);
  });

  it('should render not indexing content properly', () => {
    const wrapper = mount(
      <Link href='www.test.com' noIndex>
        Test link
      </Link>,
    );

    expect(wrapper.find('a').prop('dangerouslySetInnerHTML')).toBeDefined();
    expect(wrapper.find('a').prop('children')).not.toBeDefined();
  });

  it('should render regular (indexing) content properly', () => {
    const wrapper = mount(<Link href='www.test.com'>Test link</Link>);

    expect(wrapper.text()).toEqual('Test link');
    expect(wrapper.prop('dangerouslySetInnerHTML')).not.toBeDefined();
    expect(wrapper.prop('children')).toBeDefined();
  });

  it('render render pseudo link properly', () => {
    const wrapper = mount(<Link pseudo>Test link</Link>);

    expect(wrapper).toMatchSnapshot();
  });

  it('render render pseudo link disabled properly', () => {
    const wrapper = mount(
      <Link pseudo disabled>
        Test link
      </Link>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
