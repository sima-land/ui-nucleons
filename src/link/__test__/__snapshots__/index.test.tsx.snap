// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> calls helpers and renders correctly with external 1`] = `
<div>
  <a
    class="link basic-blue"
    data-testid="anchor"
    href="/cabinet/"
  >
    Test link
  </a>
</div>
`;

exports[`<Link /> calls helpers and renders correctly with pseudo 1`] = `
<div>
  <a
    class="link basic-gray38"
    data-testid="anchor"
    role="button"
    tabindex="0"
  >
    Test link
  </a>
</div>
`;

exports[`<Link /> calls helpers with right params and renders correctly without external 1`] = `
<div>
  <a
    class="link testClass basic-gray38"
    data-testid="anchor"
    href="/cart/"
    target="_blank"
  >
    Test link
  </a>
</div>
`;

exports[`<Link /> render render pseudo link disabled properly 1`] = `
<div>
  <a
    class="link basic-blue disabled"
    data-testid="anchor"
    role="button"
  >
    Test link
  </a>
</div>
`;

exports[`<Link /> render render pseudo link properly 1`] = `
<div>
  <a
    class="link basic-blue"
    data-testid="anchor"
    role="button"
    tabindex="0"
  >
    Test link
  </a>
</div>
`;
